src/PostHog.AI/PostHogOpenAIHandler.cs

```diff
@@ -0,0 +1,744 @@
+                        var choice = new JsonObject();
+                        var message = new JsonObject();
+                        message["role"] = "assistant";
+                        message["content"] = accumulatedResponse;
+                        choice["message"] = message;
+                        choicesArray.Add(choice);
+                    }
+
+                    if (usageNode != null)
+                    {
+                        responseNode["usage"] = usageNode;
+                    }
+
+                    _ = Task.Run(
+                        () =>
+                            CaptureEventAsync(
+                                request,
+                                requestJson,
+                                response,
+                                responseNode,
+                                stopwatch.Elapsed.TotalSeconds,
+                                eventName,
+                                null,
+                                CancellationToken.None
+                            ),
+                        CancellationToken.None
+                    );
+                }
+            );
+
+            var newContent = new StreamContent(trackingStream);
+
+            // Copy headers
+            foreach (var header in response.Content.Headers)
+            {
+                newContent.Headers.TryAddWithoutValidation(header.Key, header.Value);
+            }
+
+            response.Content = newContent;
+        }
+        else
+        {
+            stopwatch.Stop();
+            var (responseContent, responseJson) = await ReadContentAndParseJsonAsync(
+                response.Content,
+                cancellationToken,
+                ex => _logger.LogResponseContentFailure(ex)
+            );
+
+            _ = Task.Run(
+                () =>
+                    CaptureEventAsync(
+                        request,
+                        requestJson,
+                        response,
+                        responseJson,
+                        stopwatch.Elapsed.TotalSeconds,
+                        eventName,
+                        null,
+                        CancellationToken.None
+                    ),
+                CancellationToken.None
+            );
+        }
+
+        return response;
+    }
+
+    private static async Task<(string? Content, JsonNode? Json)> ReadContentAndParseJsonAsync(
+        HttpContent? content,
+        CancellationToken cancellationToken,
+        Action<Exception> onException
+    )
+    {
+        string? contentString = null;
+        JsonNode? jsonNode = null;
+
+        try
+        {
+            if (content != null)
+            {
+#pragma warning disable CA2016 // Forward cancellation token (ReadAsStringAsync has an overload with CancellationToken in .NET 5+, LoadIntoBufferAsync doesn't)
+                await content.LoadIntoBufferAsync();
+                contentString = await content.ReadAsStringAsync(cancellationToken);
```

Comment:
This buffers the content twice - once in `LoadIntoBufferAsync()` and again when `ReadAsStringAsync()` creates a string. For large responses, consider parsing JSON directly from the stream:

```csharp
using var stream = await content.ReadAsStreamAsync(cancellationToken);
jsonNode = await JsonNode.ParseAsync(stream, cancellationToken:
cancellationToken);
```